### ISSUE TITLE
[java] Int64.is() failure

### ISSUE THREAD

**Issue by kevinresol** [HaxeFoundation/haxe#5204]
```haxe
trace(Int64.is(Int64.make(1,1)));
```

fails with:

```
src/haxe/root/Main.java:31: error: unexpected type
                haxe.Log.trace.__hx_invoke2_o(0.0, ( ((long) (this1) ) instanceof java.lang.Long ), 0.0, new haxe.lang.DynamicObject(new java.lang.String[]{
"className", "fileName", "methodName"}, new java.lang.Object[]{"Main", "Main.hx", "main"}, new java.lang.String[]{"lineNumber"}, new double[]{((double) (((d
ouble) (23) )) )}));
                                                     ^
  required: reference
  found:    long
1 error
```

haxe version: a78827d
Related: https://github.com/HaxeFoundation/haxe/issues/5168

**Comment by nadako** [comment]
This is related to inlining, Java compiler complains that `instanceof` operator is used on `long` instead of `Object`. More easy-to-read example:
```haxe
import haxe.Int64;

class Main {
    static function main() {
        var l = Int64.make(1,1);
        var b = Int64.is(l);
    }
}
```
generated java code:
```java
public static void main()
{
	long this1 = ((long) (( ((long) (( ((long) (1) ) << 32 )) ) | ((long) (( ((long) (1) ) & 0xffffffffL )) ) )) );
	long l = ((long) (this1) );
	boolean b = ( l instanceof java.lang.Long );
}
```
error:
```
src\haxe\root\Main.java:38: error: unexpected type
		boolean b = ( l instanceof java.lang.Long );
		              ^
  required: reference
  found:    long
```

A stupid fix would be to add a `cast` to `Int64.is`, like this:
```haxe
return Std.is(cast val,java.lang.Long.LongClass)
```
so it would generate `( ((java.lang.Object) (l) ) instanceof java.lang.Long )` which javac wouldn't complain about, but I'll leave this to @waneck

**Comment by nadako** [comment]
Let's fix this one way or another for 3.3 (i'm not sure if this is a regression though).

**Comment by nadako** [comment]
Ooh, this is actually a duplicate (or rather, specific case) of #5168
